example_scripts/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/urfave/cli"
+)
+
+func main() {
+	color.Set(color.FgBlue, color.Italic)
+	defer color.Unset()
+	app := &cli.App{
+		Name:  "Hub XMLRPC API",
+		Usage: "helps to operate SUSE Manager or Uyuni infrastructures with one Server, called a Hub, managing several Servers.",
+	}
+	app.EnableBashCompletion = true
+	app.Commands = []*cli.Command{
+		{
+			Name:    "example",
+			Aliases: []string{"ex"},
+			Usage:   "list examples to help user start  with",
+			Subcommands: []*cli.Command{
+				{
+					Name:    "manual_login_mode",
+					Aliases: []string{"manual"},
+					Usage:   "example to show usage of manual mode",
+					Action: func(c *cli.Context) error {
+						readExampleContents("manual")
+						return nil
+					},
+				},
+				{
+					Name:    "auto_realy_login_mode",
+					Aliases: []string{"relay"},
+					Usage:   "example to show usage of relay mode",
+					Action: func(c *cli.Context) error {
+						readExampleContents("relay")
+						return nil
```

Comment:
Uhmm.... Quite weird design. :wink:  No, you don't do that. You just put one flag, like "mode" and then one global action and simply pass context there. Then inside `readExampleContents` you get it like:

```go
switch c.String("mode"):
case "relay":
    ...
default:
    cli.ShowAppHelpAndExit(c, 1)
```

Reply:
P.S. I think just a simple bash script is less overkill for this tho. :wink: